qa/tasks/cephfs/test_snapshots.py

```diff
@@ -442,3 +444,51 @@ def test_multimds_hardlink(self):
 
         self.mount_a.run_shell(["rmdir", "d1/.snap/s1"])
         self.mount_a.run_shell(["rm", "-rf", "d0", "d1"])
+
+    def delete_dir_and_snaps(self, dir_name, snaps):
+        for sno in range(1, snaps+1, 1):
+            sname = "{dir_name}/.snap/s_{sno}".format(dir_name=dir_name, sno=sno)
+            self.mount_a.run_shell(["rmdir", sname])
+        self.mount_a.run_shell(["rmdir", dir_name])
+
+    def create_dir_and_snaps(self, dir_name, snaps):
+        self.mount_a.run_shell(["mkdir", dir_name])
+        
+        # creating first 100 snapshots must be successful
+        for sno in range(1, snaps+1, 1):
+            sname = "{dir_name}/.snap/s_{sno}".format(dir_name=dir_name, sno=sno)
+            try:
+                self.mount_a.run_shell(["mkdir", sname])
+            except CommandFailedError as e:
+                # failing at the last mkdir is expected
+                if sno == snaps:
+                    log.info("failed while creating snap #{}: {}".format(sno, repr(e)))
+                    sys.exc_clear()
+
+
+    def test_mds_max_snaps_per_dir(self):
+        """
+        Test the newly introudced option named mds_max_snaps_per_dir
+        Default snaps limit is 100
+        """
+        self.fs.set_allow_new_snaps(True)
+        status = self.fs.wait_for_daemons()
+        self.create_dir_and_snaps("accounts", 101)
+        self.delete_dir_and_snaps("accounts", 100)
+
+    def test_mds_max_snaps_per_dir_2(self):
+        """
+        Test the newly introudced option named mds_max_snaps_per_dir
+        Change the default limit to test the configuration path
+        """
+        status = self.fs.wait_for_daemons()
+        self.fs.set_allow_new_snaps(True)
+        #log.info("current mds_max_snaps_per_dir: {}".format(self.ceph_cluster.get_config("mds_max_snaps_per_dir", "mds")))
+        #self.ceph_cluster.set_ceph_conf("mds.a", "mds_max_snaps_per_dir", "50")
+        self.fs.mds_asok(['config', 'set', 'mds_max_snaps_per_dir', '50'], 'a')
```

Comment:
Setting the config using the asok interface is not persistent. That's probably why it didn't work for you.

I don't think a test involving MDS failover (rank_restart) is really necessary.

Reply:
@batrick Unfortunately, `CephFSTestCase.set_conf()` isn't working for me, or maybe I don't have the correct sequence of calls. Following is the sequence that I found to be working:
```
    def test_mds_max_snaps_per_dir_2(self):
        """
        Test the newly introudced option named mds_max_snaps_per_dir
        Change the default limit to test the configuration path
        """
        self.fs.mds_asok(['config', 'set', 'mds_max_snaps_per_dir', '50'], 'a')
        self.fs.mds_asok(['config', 'set', 'mds_max_snaps_per_dir', '50'], 'b')
        self.fs.mds_asok(['config', 'set', 'mds_max_snaps_per_dir', '50'], 'c')
        self.fs.rank_restart(0)
        self.create_dir_and_snaps("accounts2", 51)
        self.delete_dir_and_snaps("accounts2", 50)
```
Somehow, a `rank_restart()` is required. I'm not too aware of the infra and its implications here.

Reply:
It's probably not necessary to use `CephFSTestCase.set_conf` because you don't need the config change to be persistent. I would use the `self.fs.rank_asok` (instead of `mds_asok`) calls for now.

The rank restart should definitely not be necessary. Please double check.

Reply:
@batrick 
With:
```
self.fs.rank_asok(['config', 'set', 'mds_max_snaps_per_dir', '50'])
```
the value does get set:
```
2019-10-08 10:17:51,190.190 INFO:tasks.cephfs.filesystem:_json_asok output: {
    "success": "mds_max_snaps_per_dir = '50' (not observed, change may require restart) "
}
```
but, no luck with that either. A `rank_restart()` doesn't help either.